### PR TITLE
Adopt glint

### PR DESCRIPTION
**Pull request by lukemelia**

### CODE REVIEW

File: demo-app/app/components/accordion/index.hbs [hunk 1]
```diff
@@ -31,8 +31,8 @@
   local-class="Accordion {{if this.isTriggerFocused "focus"}}"
   {{on "focusin" this.handleFocusin}}
   {{on "focusout" this.handleFocusout}}
-  {{on-key "ArrowUp" this.jumpToPreviousTrigger event="keyup"}}
-  {{on-key "ArrowDown" this.jumpToNextTrigger event="keyup"}}
+  {{on-key-mod "ArrowUp" this.jumpToPreviousTrigger event="keyup"}}
+  {{on-key-mod "ArrowDown" this.jumpToNextTrigger event="keyup"}}
```
Comment (nickschot) [inline demo-app/app/components/accordion/index.hbs:35]: Just a double check, but is this change correct? Does this modifier exist with this name?

File: addon/addon/models/sprite-tree.ts [hunk 2]
```diff
@@ -23,7 +23,7 @@ export interface IContext {
   appendOrphan(spriteOrElement: Sprite): void;
   clearOrphans(): void;
   args: {
-    use?(changeset: Changeset): AnimationDefinition;
+    use?(changeset: Changeset): AnimationDefinition | Promise<void>;
```
Comment (nickschot) [inline addon/addon/models/sprite-tree.ts]: The `Promise<void>` seems like it's based on the old system/api. We won't support this in the future, but right now a lot of the demos are still using the old API. Just something to keep in mind I guess (as we're unlikely to refactor all demos immediately).

File: demo-app/app/components/accordion/index.hbs (hunk 1 above)
Comment (lukemelia) [reply demo-app/app/components/accordion/index.hbs:35]: Ah. yes, this warrants an explanation. The way glint deals with non-strict templates is to let you add types in a global registry. There is a single registry that includes components and helpers and modifiers. ember-keyboard has both a helper named `on-key` and a modifier named `on-key`. This is impossible to work with as-is with the system I just described. As a workaround, in this PR, we re-export the `on-key-mod` modifier as `on-key-mod` and add that to the registry.

Comment (nickschot) [reply demo-app/app/components/accordion/index.hbs:35]: Ahhh! That explains indeed.

File: addon/addon/components/animation-context/index.ts [hunk 3]
```diff
@@ -18,17 +18,20 @@ const { VOLATILE_TAG, consumeTag } =
...
+  id?: string;
+  use: ((changeset: Changeset) => AnimationDefinition) | undefined;
+}
+
+interface Signature {
+  Element: HTMLDivElement;
+  Args: AnimationContextArgs;
   Blocks: {
-    default: [];
+    default: [AnimationContextComponent];
```
Comment (nickschot) [inline addon/addon/components/animation-context/index.ts:29]: No idea why we still yield this, but I don't think we have any use case for it. We can remove it (either in this PR or immediately after we merge this I suppose).

**Review by nickschot — APPROVED** [review]

File: addon/addon/components/animation-context/index.ts (hunk 3 above)
Comment (lukemelia) [reply addon/addon/components/animation-context/index.ts:29]: I'll leave that to you after this PR. Having type-checking through the template layer reveals unintended APIs. 👍

**Pull request merged**